src/com/twu/biblioteca/Book.java

```diff
@@ -0,0 +1,35 @@
+package com.twu.biblioteca;
+
+public class Book {
+    private final String title;
+    private final String author;
+    private final int yearPublished;
+    private boolean checkedOut;
+
+    public Book(String title, String author, int yearPublished, boolean checkedOut) {
```

Comment:
Does IntelliJ give you any messages about the `checkedOut` parameter here?

Reply:
It says that it is never used. I'm not sure why it's saying that.

Reply:
I'm on the side of IntelliJ here :-)  If you look closely at how you are using the first three params compared to the last one, it might become clearer.  

Because of the static nature of the Java language (and the maturity of IntelliJ), paying close attention to all those IntelliJ warnings is generally a very good idea.  They are often soft pointers to likely bugs.

Reply:
I'm not following the convention. It should be:
this.checkedOut = checkedOut;